server: TestClusterVersionUpgrade failed

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9286267?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9286267?buildTab=artifacts#/) on release-23.1 @ [3551dc089ac3efcb3772df2109164dcb01e79d8f](https://github.com/cockroachdb/cockroach/commits/3551dc089ac3efcb3772df2109164dcb01e79d8f):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade867280458
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-90, should be 1000022.2-94
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade867280458
--- FAIL: TestClusterVersionUpgrade (48.63s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9309104?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9309104?buildTab=artifacts#/) on release-23.1 @ [a34af06d8e7230e4052db80d55da21a06013f81b](https://github.com/cockroachdb/cockroach/commits/a34af06d8e7230e4052db80d55da21a06013f81b):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade1766119000
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-90, should be 1000022.2-94
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade1766119000
--- FAIL: TestClusterVersionUpgrade (48.36s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9332997?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9332997?buildTab=artifacts#/) on release-23.1 @ [aec78f33d45a8376a0ecec885688bae60dbfb85c](https://github.com/cockroachdb/cockroach/commits/aec78f33d45a8376a0ecec885688bae60dbfb85c):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade638750292
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-90, should be 1000022.2-94
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade638750292
--- FAIL: TestClusterVersionUpgrade (48.52s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9356604?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9356604?buildTab=artifacts#/) on release-23.1 @ [7e72aae900c3ff4b44f1643c2d7ba55fbb2cbe23](https://github.com/cockroachdb/cockroach/commits/7e72aae900c3ff4b44f1643c2d7ba55fbb2cbe23):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade2602651846
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-82, should be 1000022.2-94
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade2602651846
--- FAIL: TestClusterVersionUpgrade (48.43s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9376951?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9376951?buildTab=artifacts#/) on release-23.1 @ [5d5abe704b0551556c35d51aa673d3162f493aa8](https://github.com/cockroachdb/cockroach/commits/5d5abe704b0551556c35d51aa673d3162f493aa8):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade2495400002
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-70, should be 1000022.2-96
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade2495400002
--- FAIL: TestClusterVersionUpgrade (48.72s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9386302?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9386302?buildTab=artifacts#/) on release-23.1 @ [63b683eab3e0ebfff812bb54fdac936c101d9383](https://github.com/cockroachdb/cockroach/commits/63b683eab3e0ebfff812bb54fdac936c101d9383):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade3759419551
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-74, should be 1000022.2-98
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade3759419551
--- FAIL: TestClusterVersionUpgrade (49.03s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9396913?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9396913?buildTab=artifacts#/) on release-23.1 @ [63b683eab3e0ebfff812bb54fdac936c101d9383](https://github.com/cockroachdb/cockroach/commits/63b683eab3e0ebfff812bb54fdac936c101d9383):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade3289630894
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-74, should be 1000022.2-98
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade3289630894
--- FAIL: TestClusterVersionUpgrade (48.96s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9444359?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9444359?buildTab=artifacts#/) on release-23.1 @ [e847e4358c004d4a8977ae26749768ad1d417fba](https://github.com/cockroachdb/cockroach/commits/e847e4358c004d4a8977ae26749768ad1d417fba):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade1590784602
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-72, should be 1000022.2-98
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade1590784602
--- FAIL: TestClusterVersionUpgrade (49.14s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9465374?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9465374?buildTab=artifacts#/) on release-23.1 @ [b432e8c20339de5cfa7c811a9ee6f5dc98d15a1e](https://github.com/cockroachdb/cockroach/commits/b432e8c20339de5cfa7c811a9ee6f5dc98d15a1e):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade485808879
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 1000022.2-52, should be 1000022.2-98
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade485808879
--- FAIL: TestClusterVersionUpgrade (49.79s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9485637?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9485637?buildTab=artifacts#/) on release-23.1 @ [571a1744c72415894dc0e742bd89e74938ea71f4](https://github.com/cockroachdb/cockroach/commits/571a1744c72415894dc0e742bd89e74938ea71f4):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade1245085385
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 22.2-66, should be 22.2-98
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade1245085385
--- FAIL: TestClusterVersionUpgrade (49.27s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #79162 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-21.2]
- #78126 server: TestClusterVersionUpgrade failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9501691?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9501691?buildTab=artifacts#/) on release-23.1 @ [de239a7438f44d382c9aefceb65d9c39911dabd2](https://github.com/cockroachdb/cockroach/commits/de239a7438f44d382c9aefceb65d9c39911dabd2):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade3817485600
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 22.2-72, should be 22.2-98
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade3817485600
--- FAIL: TestClusterVersionUpgrade (49.16s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9510568?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9510568?buildTab=artifacts#/) on release-23.1 @ [de239a7438f44d382c9aefceb65d9c39911dabd2](https://github.com/cockroachdb/cockroach/commits/de239a7438f44d382c9aefceb65d9c39911dabd2):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade576472633
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 22.2-70, should be 22.2-98
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade576472633
--- FAIL: TestClusterVersionUpgrade (49.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestClusterVersionUpgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9519417?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9519417?buildTab=artifacts#/) on release-23.1 @ [2f96695f75b07c872ec5f146acc1fa198135768f](https://github.com/cockroachdb/cockroach/commits/2f96695f75b07c872ec5f146acc1fa198135768f):


```
=== RUN   TestClusterVersionUpgrade
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade2436920118
    test_log_scope.go:79: use -show-logs to present logs inline
    version_cluster_test.go:259: condition failed to evaluate within 45s: cluster version is still 22.2-78, should be 22.2-98
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestClusterVersionUpgrade2436920118
--- FAIL: TestClusterVersionUpgrade (48.80s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterVersionUpgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/version_cluster_test.go
+++ b/pkg/server/version_cluster_test.go
@@ -17,6 +17,7 @@ import (
 	"path/filepath"
 	"strconv"
 	"testing"
+	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/clusterversion"
@@ -256,16 +257,16 @@ func TestClusterVersionUpgrade(t *testing.T) {
 	}
 
 	// Check the cluster version is bumped to newVersion.
-	testutils.SucceedsSoon(t, func() error {
+	testutils.SucceedsWithin(t, func() error {
 		if version := tc.getVersionFromSelect(0); version != newVersion.String() {
 			return errors.Errorf("cluster version is still %s, should be %s", version, newVersion)
 		}
 		return nil
-	})
+	}, 3*time.Minute)
 	curVersion = tc.getVersionFromSelect(0)
 	isNoopUpdate := curVersion == newVersion.String()
 
-	testutils.SucceedsSoon(t, func() error {
+	testutils.SucceedsWithin(t, func() error {
 		for i := 0; i < tc.NumServers(); i++ {
 			st := tc.Servers[i].ClusterSettings()
 			v := st.Version.ActiveVersion(ctx)
@@ -279,14 +280,14 @@ func TestClusterVersionUpgrade(t *testing.T) {
 			}
 		}
 		return nil
-	})
+	}, 3*time.Minute)
 
 	exp := newVersion.String()
 
 	// Read the versions from the table from each node. Note that under the
 	// hood, everything goes to the lease holder and so it's pretty much
 	// guaranteed that they all read the same, but it doesn't hurt to check.
-	testutils.SucceedsSoon(t, func() error {
+	testutils.SucceedsWithin(t, func() error {
 		for i := 0; i < tc.NumServers(); i++ {
 			if version := tc.getVersionFromSelect(i); version != exp {
 				return errors.Errorf("%d: incorrect version %q (wanted %s)", i, version, exp)
@@ -296,13 +297,13 @@ func TestClusterVersionUpgrade(t *testing.T) {
 			}
 		}
 		return nil
-	})
+	}, 3*time.Minute)
 
 	// Now check the Settings.Version variable. That is the tricky one for which
 	// we "hold back" a gossip update until we've written to the engines. We may
 	// have to wait a bit until we see the new version here, even though it's
 	// already in the table.
-	testutils.SucceedsSoon(t, func() error {
+	testutils.SucceedsWithin(t, func() error {
 		for i := 0; i < tc.NumServers(); i++ {
 			vers := tc.Servers[i].ClusterSettings().Version.ActiveVersion(ctx)
 			if v := vers.String(); v == curVersion {
@@ -315,7 +316,7 @@ func TestClusterVersionUpgrade(t *testing.T) {
 			}
 		}
 		return nil
-	})
+	}, 3*time.Minute)
 
 	// Since the wrapped version setting exposes the new versions, it must
 	// definitely be present on all stores on the first try.

